clarity/src/vm/functions/sequences.rs

```diff
@@ -383,3 +383,88 @@ pub fn special_slice(
     }?;
     Ok(sliced_seq)
 }
+
+pub fn special_replace_at(
+    args: &[SymbolicExpression],
+    env: &mut Environment,
+    context: &LocalContext,
+) -> Result<Value> {
+    check_argument_count(3, args)?;
+
+    let seq = eval(&args[0], env, context)?;
+    let seq_type = TypeSignature::type_of(&seq);
+    let (expected_elem_type, seq_is_list_type) =
+        if let TypeSignature::SequenceType(seq_subtype) = seq_type.clone() {
+            (seq_subtype.unit_type(), seq_subtype.is_list_type())
+        } else {
+            return Err(CheckErrors::ExpectedSequence(seq_type).into());
+        };
+    let index_val = eval(&args[1], env, context)?;
+    let new_element = eval(&args[2], env, context)?;
+
+    if expected_elem_type != TypeSignature::NoType && !expected_elem_type.admits(&new_element) {
+        return Err(CheckErrors::TypeValueError(expected_elem_type, new_element).into());
+    }
+
+    let index = if let Value::UInt(index) = index_val {
+        index
+    } else {
+        return Err(CheckErrors::TypeValueError(TypeSignature::UIntType, index_val).into());
+    };
+
+    match seq {
+        // todo: update the ClarityCostFunction once the Clarity2 related cost functions are implemented.
```

Comment:
Can you open an issue for this and tag it as `stacks-2.1` so we don't forget it?